src/ProxyManager/ProxyGenerator/LazyLoadingGhost/MethodGenerator/MagicUnset.php

```diff
@@ -126,30 +127,40 @@ public function __construct(
     ) {
         parent::__construct($originalClass, '__unset', [new ParameterGenerator('name')]);
 
-        $override = $originalClass->hasMethod('__unset');
+        $existingMethod = $originalClass->hasMethod('__unset') ? $originalClass->getMethod('__unset') : null;
 
-        $this->setDocBlock(($override ? "{@inheritDoc}\n" : '') . '@param string $name');
+        $this->setDocBlock(($existingMethod ? "{@inheritDoc}\n" : '') . '@param string $name');
 
-        $parentAccess = 'return parent::__unset($name);';
-
-        if (! $override) {
-            $parentAccess = PublicScopeSimulator::getPublicAccessSimulationCode(
+        $parentAccess = $existingMethod
+            ? ProxiedMethodReturnExpression::generate('parent::__unset($name)', $existingMethod)
+            : PublicScopeSimulator::getPublicAccessSimulationCode(
+                $existingMethod,
                 PublicScopeSimulator::OPERATION_UNSET,
                 'name'
             );
-        }
-
-        $this->setBody(sprintf(
-            $this->callParentTemplate,
-            '$this->' . $initializerProperty->getName() . ' && $this->' . $callInitializer->getName()
-            . '(\'__unset\', array(\'name\' => $name));',
-            $publicProperties->getName(),
-            $protectedProperties->getName(),
-            $protectedProperties->getName(),
-            $privateProperties->getName(),
-            $privateProperties->getName(),
-            $privateProperties->getName(),
-            $parentAccess
+
+        $symbols = [
+            '%init'                 => '$this->' . $initializerProperty->getName()
```

Comment:
I should hunt down all the `sprintf()` and replace them with this syntax: much easier to follow.